packages/client/src/components/form/LinkButton.tsx

```diff
@@ -56,13 +61,34 @@ export const LinkButtonField = ({
   const trigger = fields.find(
     (f) => f.name === fieldDefinition.options.callback.trigger
   )!
-  const onChange: Parameters<typeof useHttp>[1] = ({
+  const { declarationId = '', pageId: section } = useParams()
+  const declaration = useDeclaration(declarationId)
+  const dispatch = useDispatch()
+  const userId = useSelector(getUserDetails)?.id
+  const onChange: Parameters<typeof useHttp>[1] = async ({
     data,
     error,
     loading
   }) => {
     setFieldValue(trigger.name, { loading, data, error } as IFormFieldValue)
     if (data || error) {
+      if (section && declaration && userId) {
+        const updatedFormData =
+          FormSectionComponent.getUpdatedValuesAfterDependentFieldEvaluation(
+            form,
+            fields,
+            trigger.name,
+            { loading, data, error } as IFormFieldValue,
+            { config, draft, user }
+          )
+        writeDeclarationByUserWithoutStateUpdate(userId, {
+          ...declaration,
+          data: {
+            ...declaration.data,
+            [section]: merge(declaration.data[section], updatedFormData)
+          }
+        })
+      }
```

Comment:
Maybe a comment here to help translating to Events v2 why this is needed

Reply:
Thanks @naftis, I am on it now

Reply:
done https://github.com/opencrvs/opencrvs-core/pull/9283/commits/2055ffea9c9cbfcc3971e5c7dd852ae8ae2062ef